test/JDBC/expected/BABEL-4312.out

```diff
@@ -0,0 +1,275 @@
+        Index Cond: (b <= 'sss'::"varchar")
+~~END~~
+
+SELECT TOP 1 b FROM babel_4312_tbl WHERE b > 'sss' ORDER BY b
+go
+~~START~~
+text
+Query Text: SELECT TOP 1 b FROM babel_4312_tbl WHERE b > 'sss' ORDER BY b
+Limit  (cost=0.14..2.82 rows=1 width=32)
+  ->  Index Only Scan using babel_4312_idx2babel_4312_tbl232f4019e750f8cae4fbedfbd7c85159 on babel_4312_tbl  (cost=0.14..8.19 rows=3 width=32)
+        Index Cond: (b > 'sss'::"varchar")
+~~END~~
+
+
+-- Reset
+SET babelfish_showplan_all OFF
+go
+SELECT set_config('enable_seqscan', 'on', false)
+go
+~~START~~
+text
+on
+~~END~~
+
+
+DROP INDEX babel_4312_idx1 ON babel_4312_tbl
+go
+DROP INDEX babel_4312_idx2 ON babel_4312_tbl
+go
+
+-- DESC index (DESC NULLS LAST)
+CREATE INDEX babel_4312_idx1 ON babel_4312_tbl (a DESC)
+go
+CREATE INDEX babel_4312_idx2 ON babel_4312_tbl (b DESC)
+go
+
+SELECT TOP 1 a FROM babel_4312_tbl WHERE a <= 5 ORDER BY a DESC
+go
+~~START~~
+int
+5
+~~END~~
+
+SELECT TOP 1 a FROM babel_4312_tbl WHERE a > 5 ORDER BY a ASC
+go
+~~START~~
+int
+6
+~~END~~
+
+SELECT TOP 1 a FROM babel_4312_tbl WHERE a <= 5 ORDER BY a
+go
+~~START~~
+int
+1
+~~END~~
+
+SELECT TOP 1 a FROM babel_4312_tbl WHERE a > 5 ORDER BY a DESC
+go
+~~START~~
+int
+8
+~~END~~
+
+
+SELECT TOP 1 b FROM babel_4312_tbl WHERE b <= 'sss' ORDER BY b DESC
+go
+~~START~~
+varchar
+pqr
+~~END~~
+
+SELECT TOP 1 b FROM babel_4312_tbl WHERE b > 'sss' ORDER BY b
+go
+~~START~~
+varchar
+stu
+~~END~~
+
+SELECT TOP 1 b FROM babel_4312_tbl WHERE b <= 'sss' ORDER BY b
+go
+~~START~~
+varchar
+abc
+~~END~~
+
+SELECT TOP 1 b FROM babel_4312_tbl WHERE b > 'sss' ORDER BY b DESC
+go
+~~START~~
+varchar
+xyz
+~~END~~
+
+
+-- Check query plans, all aggregations should be optimized
+-- into LIMIT + index scan.
+SELECT set_config('enable_seqscan', 'off', false)
+go
+~~START~~
+text
+off
+~~END~~
+
+SET babelfish_showplan_all ON
+go
+
+SELECT TOP 1 a FROM babel_4312_tbl WHERE a <= 5 ORDER BY a DESC
+go
+~~START~~
+text
+Query Text: SELECT TOP 1 a FROM babel_4312_tbl WHERE a <= 5 ORDER BY a DESC
+Limit  (cost=0.14..2.82 rows=1 width=4)
+  ->  Index Only Scan using babel_4312_idx1babel_4312_tbl2e82eb34f45e1b440ddc74400aafd09c on babel_4312_tbl  (cost=0.14..8.19 rows=3 width=4)
+        Index Cond: (a <= 5)
+~~END~~
+
+SELECT TOP 1 a FROM babel_4312_tbl WHERE a > 5 ORDER BY a ASC
+go
+~~START~~
+text
+Query Text: SELECT TOP 1 a FROM babel_4312_tbl WHERE a > 5 ORDER BY a ASC
+Limit  (cost=0.14..2.82 rows=1 width=4)
+  ->  Index Only Scan Backward using babel_4312_idx1babel_4312_tbl2e82eb34f45e1b440ddc74400aafd09c on babel_4312_tbl  (cost=0.14..8.19 rows=3 width=4)
+        Index Cond: (a > 5)
+~~END~~
+
+
+SELECT TOP 1 b FROM babel_4312_tbl WHERE b <= 'sss' ORDER BY b DESC
+go
+~~START~~
+text
+Query Text: SELECT TOP 1 b FROM babel_4312_tbl WHERE b <= 'sss' ORDER BY b DESC
+Limit  (cost=0.14..2.82 rows=1 width=32)
```

Comment:
pls set select set_config('babelfishpg_tsql.explain_costs', 'off', false) , when explain plan, to make the test less flaky

Reply:
Done in the latest revision.